### PR TITLE
Feat/en 3075 elrond core vm integration and bugfixing

### PR DESCRIPTION
**Pull request by ccorcoveanu**
This first PR for this issue contains the changes/bugfixes in order to correctly run a smart contract call in a multi-shard setup with observers and a proxy.
For this issue to be closed, another PR will follow containing unit tests covering this changes

### CODE REVIEW

File: dataRetriever/resolvers/transactionResolver.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func (txRes *TxResolver) resolveTxRequestByHashArray(hashesBuff []byte, pid p2p.
 		if err != nil {
 			//it might happen to error on a tx (maybe it is missing) but should continue
 			// as to send back as many as it can
-			log.Debug(err.Error())
+			log.Info(err.Error())
```
Comment (sasurobert) [inline dataRetriever/resolvers/transactionResolver.go]: put log.Debug back

File: integrationTests/vm/iele/vmDeploy_test.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ func TestVmDeployWithTransferAndGasShouldDeploySCCode(t *testing.T) {
 	_, err = accnts.Commit()
 	assert.Nil(t, err)
 
-	expectedBalance := big.NewInt(99978648)
+	expectedBalance := big.NewInt(99922044)
```
Comment (sasurobert) [inline integrationTests/vm/iele/vmDeploy_test.go]: change the gas model in tests as well.

File: integrationTests/vm/testInitializer.go [hunk 3]
```diff
@@ -102,7 +101,7 @@ func CreateOneSCExecutorMockVM(accnts state.AccountsAdapter) vmcommon.VMExecutio
 func CreateVMAndBlockchainHook(accnts state.AccountsAdapter) (vmcommon.VMExecutionHandler, *hooks.VMAccountsDB) {
 	blockChainHook, _ := hooks.NewVMAccountsDB(accnts, addrConv)
 	cryptoHook := hooks.NewVMCryptoHook()
-	vm := endpoint.NewElrondIeleVM(blockChainHook, cryptoHook, ielecommon.Danse)
+	vm := endpoint.NewElrondIeleVM(blockChainHook, cryptoHook, endpoint.ElrondDefault)
```
Comment (sasurobert) [inline integrationTests/vm/testInitializer.go]: ElrondTestnet

File: process/coordinator/process.go [hunk 4]
```diff
@@ -132,7 +132,7 @@ func (tc *transactionCoordinator) RequestBlockTransactions(body block.Body) {
 	wg.Add(len(separatedBodies))
 
 	for key, value := range separatedBodies {
-		go func() {
+		go func(key block.Type, value block.Body) {
```
Comment (sasurobert) [inline process/coordinator/process.go]: little bit confusing to have the same variables in call and parameters as well.

File: dataRetriever/resolvers/transactionResolver.go (hunk 1 above)
Comment (ccorcoveanu) [reply dataRetriever/resolvers/transactionResolver.go]: done

File: integrationTests/vm/iele/vmDeploy_test.go (hunk 2 above)
Comment (ccorcoveanu) [reply integrationTests/vm/iele/vmDeploy_test.go]: done

File: integrationTests/vm/testInitializer.go (hunk 3 above)
Comment (ccorcoveanu) [reply integrationTests/vm/testInitializer.go]: done

**Pull request merged**